### PR TITLE
[DNM] s3 lock service review

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Management/S3Lock.cpp [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    std::unordered_map<std::string, DataFileMutexPtr>::iterator it;
+    {
+        std::unique_lock lock(file_latch_map_mutex);
+        it = file_latch_map.find(data_file_name);
+        if (it == file_latch_map.end())
+        {
+            it = file_latch_map.emplace(data_file_name, std::make_shared<DataFileMutex>()).first;
+        }
+    }
+    auto & file_lock = it->second;
```
Comment (JaySon-Huang) [inline dbms/src/Flash/Management/S3Lock.cpp:145]:
```suggestion
    DataFileMutexPtr file_lock;
    {
        std::unique_lock lock(file_latch_map_mutex);
        it = file_latch_map.find(data_file_name);
        if (it == file_latch_map.end())
        {
            it = file_latch_map.emplace(data_file_name, std::make_shared<DataFileMutex>()).first;
        }
        file_lock = it->second;
        file_lock->addRefCount();
    }
```
Should only return the shared_ptr to upper scope, and should add ref count under the unique_lock of file_latch_map_mutex

File: dbms/src/Flash/Management/S3Lock.cpp [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
```
Comment (JaySon-Huang) [inline dbms/src/Flash/Management/S3Lock.cpp:211]: ditto

Comment (Lloyd-Pottiger) [reply dbms/src/Flash/Management/S3Lock.cpp]: fixed

**Comment by JaySon-Huang** [comment]
closed by https://github.com/pingcap/tiflash/pull/6874 to the master branch

**Pull request closed without merging**